File: src/go-backend/internal/http/model/stream.go
package model

// URL params
type StreamRequest struct {
	ServiceTag    string
	ShowID        string
	SeasonNumber  int
	EpisodeNumber int
	StreamType    string

	IsPlaylist bool
	IsMedia    bool

	Playlist StreamPlaylist
	Media    StreamMedia
}

type StreamPlaylist struct {
	FileName string
}

type StreamMedia struct {
	URL string

	Dash StreamDashMedia
}

type StreamDashMedia struct {
	ID   string
	Type string
}

// Stream Response
type Stream []byte

File: src/frontend/src/lib/api/stream.ts
import { getAPIURL } from "$lib/config/url";
import type { StreamRequest } from "$lib/types/stream";

export const getStreamURL = (request: StreamRequest): string => {
    return `${getAPIURL()}/stream/${request.ServiceTag}/${request.ShowID}/${request.SeasonNumber}/${request.EpisodeNumber}/${request.StreamType}/${request.StreamFileName}`;
};
